apis/v1alpha1/httproute_types.go

```diff
@@ -427,64 +447,111 @@ type HTTPRequestHeaderFilter struct {
 	// TODO
 }
 
-// HTTPRequestMirrorFilter defines configuration for the
-// RequestMirror filter.
+// HTTPRequestMirrorFilter defines configuration for the RequestMirror filter.
 type HTTPRequestMirrorFilter struct {
-	// TargetRef is an object reference to forward matched requests to.
-	// The resource may be "services" (omit or use the empty string for the
-	// group), or an implementation may support other resources (for
-	// example, resource "myroutetargets" in group "networking.acme.io").
-	// Omitting or specifying the empty string for both the resource and
-	// group indicates that the resource is "services".  If the referent
-	// cannot be found, the "InvalidRoutes" status condition on any Gateway
-	// that includes the HTTPRoute will be true.
+	// ServiceName refers to the name of the Service to mirror matched requests
+	// to. When specified, this takes the place of BackendRef. If both
+	// BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this
+	// Route to true.
 	//
-	// Support: Core (Kubernetes Services)
-	// Support: Implementation-specific (Other resource types)
+	// Support: Core
 	//
-	TargetRef ForwardToTargetObjectReference `json:"targetRef"`
+	// +optional
+	// +kubebuilder:validation:MaxLength=253
+	ServiceName *string `json:"serviceName,omitempty"`
 
-	// TargetPort specifies the destination port number to use for the TargetRef.
-	// If unspecified and TargetRef is a Service object consisting of a single
-	// port definition, that port will be used. If unspecified and TargetRef is
-	// a Service object consisting of multiple port definitions, an error is
-	// surfaced in status.
+	// BackendRef is a local object reference to mirror matched requests to. If
+	// both BackendRef and ServiceName are specified, ServiceName will be given
+	// precedence. If the referent cannot be found, controllers must set the
+	// "InvalidRoutes" status condition on any Gateway that includes this Route
+	// to true.
+	//
+	// Support: Extended
+	//
+	// +optional
+	BackendRef LocalObjectReference `json:"backendRef,omitempty"`
+
+	// Port specifies the destination port number to use for the backend. If
+	// unspecified and a Service object consisting of a single port definition
+	// is the backend, that port will be used. If unspecified and the backend is
+	// a Service object consisting of multiple port definitions, controllers
+	// must set the "InvalidRoutes" status condition on any Gateway that
+	// includes this Route to true.
 	//
 	// Support: Core
 	//
 	// +optional
-	TargetPort *TargetPort `json:"targetPort,omitempty"`
+	// +kubebuilder:validation:Minimum=1
```

Comment:
Same as https://github.com/kubernetes-sigs/service-apis/pull/353/files#r496895810

Reply:
I agree that we don't need to specify `MinItems` or `MinLength` for optional fields, but a `Minimum` is still important here so we ensure that if users do set this value it's not 0 or negative.